pyth/tests/test_update_price.py

```diff
@@ -0,0 +1,105 @@
+
+    messageIds = itertools.count()
+
+    # Use a single websocket connection for the entire test, as batching is done per-user
+    async with websockets.connect('ws://localhost:8910/') as ws:
+
+        async def update_price(account, price, conf, status):
+            msg = jrpc_req(
+                method='update_price',
+                params={
+                    'account': account,
+                    'price': price,
+                    'conf': conf,
+                    'status': status,
+                })
+
+            await send(msg)
+            resp = await recv()
+            assert resp['result'] == 0
+
+        async def get_product(account):
+            msg = jrpc_req(method='get_product', params={
+                'account': account
+            })
+
+            await send(msg)
+            result = await recv()
+
+            return result
+
+        def jrpc_req(method=None, params=None):
+            return {
+                'jsonrpc': '2.0',
+                'method': method,
+                'params': params,
+                'id': next(messageIds)
+            }
+
+        async def send(msg):
+            print("--- sending message ---")
+            print(msg)
+            await ws.send(json.dumps(msg))
+
+        async def recv():
+            data = await ws.recv()
+            msg = json.loads(data)
+            print("----- received message -----")
+            print(msg)
+            return msg
+
+        def get_publisher_acc(product_acc):
+            assert len(product_acc['result']['price_accounts']) == 1
+            price_acc = product_acc['result']['price_accounts'][0]
+
+            assert len(price_acc['publisher_accounts']) == 1
+            return price_acc['publisher_accounts'][0]
+
+        # Generate new values for this test
+        new_values = {
+            product: {
+                'price':random.randint(1, 150),
+                'conf': random.randint(1, 20),
+                'status': 'trading',
+            } for product in PRODUCTS.keys()
+        }
+
+        # Update the values of the products
+        for product in PRODUCTS.keys():
+            await update_price(
+                pyth_init_price[product],
+                new_values[product]['price'],
+                new_values[product]['conf'],
+                new_values[product]['status'])
+
+        time.sleep(60)
+
+        # Crank the products
+        for product in PRODUCTS.keys():
+            await update_price(
+                pyth_init_price[product],
+                1,
+                1,
+                'trading')
+
+        time.sleep(60)
+
+        # Check that the price has been updated
+        for product in PRODUCTS.keys():
+            product_acc = await get_product(pyth_init_product[product])
+            publisher_acc = get_publisher_acc(product_acc)
+
+            assert publisher_acc['price'] == new_values[product]['price']
```

Comment:
I think it would be good to test the end-to-end API somewhere. Let's use pythd exactly as a publisher would: register for the price callbacks and then submit updates in response. This is probably not easy to do in a unit test, but could be a mock publisher or something. 

(That doesn't need to happen as part of this PR.)

Reply:
> why is this the value in new_values and not 1?

The second update is just to crank the prices.